sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInYieldShouldBeWrapped.cs

```diff
@@ -81,5 +64,25 @@ public static IEnumerable<string> Get(int val)
 
             yield return "test";
         }
+
+        public static IEnumerable<TSource> TakeWhile2<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
```

Comment:
Maybe change the method name to _TakeWhileWithLocalFunction_ to make it immediately obvious what case this method is testing.